### DISCUSSION TITLE
Replace the tasks and tabs system with an integrated in-app workspace bar

### DISCUSSION THREAD

**Discussion by EmmanuelMess** [TeamAmaze/AmazeFileManager#3023] [Ideas]
Replace the tasks and tabs system with an integrated in-app task bar.

This a small draft I created, the idea is that a workspace can be any Editor (database editor, text editor, etc) or a folder.

![IMG_20211114_133349](https://user-images.githubusercontent.com/10991116/141689892-b7303752-c6fb-4578-965c-0812bd235f6f.png)
![IMG_20211114_133403](https://user-images.githubusercontent.com/10991116/141689981-5560b830-3e91-48da-8bf0-34093882a47f.png)

**Comment by EmmanuelMess** [comment]
I am leaning more towards using a simple miniFAB for opening the workspaces bottom bar

![IMG_20211116_204059](https://user-images.githubusercontent.com/10991116/142083866-dde35f24-6f66-4b9f-9d5f-c7e5266a0c38.png)

**Comment by shuvashish76** [comment]
suggestion:
![20211117_192933](https://user-images.githubusercontent.com/48226312/142245759-123031e6-4215-40fd-ab33-b14e38dd1bbd.jpeg)

What I changed.
- Delete button for each workspace
- Long click select multiple workspace & delete them from top right corner delete button (that delete button should appear only when you pick multiple workspaces)
- No of active workspaces (on top right corner)
- Click on that "number" to view workspaces in list view (just like android chrome browser).

Other features idea
- Option to freely move workspaces (just like android browser tabs)
- Option to pin workspace (IMO that will be too much as we've sidebar_bookmarks, shortcuts. Just an idea)
- Option to remember if Amaze used workspace in default view or list view.
- Open folders in new workspace just like #2744 
- Choose your default folder for new default workspace (each time you click + to create new workspace) from Amaze settings.

**Reply by shuvashish76** [reply]
The "+" button only next to laast workspace isn't an good idea. If you opened 10 workspaces then you need to scroll all the way down to 10th workspace to open up your 11the workspace.

> Long click select multiple workspace & delete them from top right corner delete button (that delete button should appear only when you pick multiple workspaces)

Otherwise there should be a "+" button there for creating new workspace.

**Reply by EmmanuelMess** [reply]
* The number of workspaces is redundant.
* The delete button is too small. I was planning to do that with a swipe.
* The option to move workspaces is not needed for now. 
* Pinning workspaces will make them a consistently long list
* Open folders in new workspace is the central piece of this feature
